server/routes/gcp/projects.py

```
@@ -162,22 +164,45 @@ def _check_project_iam(crm_service, pid, name, member_sa, root_project):
 def _oauth_mode_project_list(credentials, sa_email, root_project):
     """Build project list for OAuth mode by enumerating projects + IAM."""
+    from concurrent.futures import ThreadPoolExecutor, as_completed
+    from googleapiclient._helpers import positional
+
     projects = get_project_list(credentials)
-    crm_service = build('cloudresourcemanager', 'v1', credentials=credentials)
     member_sa = f"serviceAccount:{sa_email}"
 
+    items = [(p.get('projectId'), p.get('name', p.get('projectId'))) for p in projects if p.get('projectId')]
+
+    def check_one(pid_name):
```

Ruff (0.15.13)

[warning] Missing return type annotation for private function `check_one`

(ANN202)